### PR TITLE
Add mobile menu and fix some mobile stylings

### PR DESCRIPTION
**Pull request by ktbee**
@evmiguel @rwaldron this PR fixes our biggest mobile styling issues, namely no mobile friendly menu and uncentered scenario cards.

Proposed changes:
- Style changes for centering scenarios on the front end
- Add DesktopNav and MobileNav components. The mobile one includes a button for expanding/closing on click using handy dandy React Hooks!
- Refactor the menu to have most of its components in separate files 

Screen Shot 2019-11-20 at 5 29 49 PM

### CODE REVIEW

File: client/routes/MobileNav.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+const MobileNav = () => {
+    const [menuExpanded, setMenuExpanded] = useState(false);
+    return (
+        <React.Fragment>
+            <Button
+                icon
+                fluid
+                labelPosition="left"
+                basic
+                color="black"
```
Comment (rwaldron) [inline client/routes/MobileNav.jsx]: I'm not sure why these buttons have the dark black border—i've always wondered about that that 🤷‍♂ It's especially strange now that menu is default on desktop and then has this harsh black border on mobile. Can we just ditch hard-coded color on buttons and leave them default?

**Comment by rwaldron** [comment]
I'm trying to think of ways we can do this without needing two nearly identical sets of menu code. If mobile view is only needed for participant runs, then the menu will only have 2 or three items in it, right? In that case, couldn't we just add the `stackable` attribute to the `Navigation` component in routes/index.js?

```js
const Navigation = () => {
    return (
        <Menu className="nav" stackable>
```

**Comment by rwaldron** [comment]
Per my comment above: 

image


Keep in mind that only two of those items would show for  a logged in user, and only one for an anonymous user.

File: client/routes/MobileNav.jsx (hunk 1 above)
Comment (ktbee) [reply client/routes/MobileNav.jsx]: I like it, but don't care particularly strongly enough to say they should be black. I had also worried about color contrast with Semantic UI's default button styling, but I just checked it and it seems to be fine according to the WAVE accessibility extension. We can have it be the default if you want.

**Comment by ktbee** [comment]
I see your point about not having two sets of menu's to maintain @rwaldron. However I think the stackable menu takes up too much screen real estate on mobile for the scenarios list. If you're not logged in, you will see three things: "Teacher Moments," "Curated", "Log In". That's a lot of space to lose on small screens. 

I'll think of some solutions and see what you think.

**Comment by ktbee** [comment]
@rwaldron do you mind taking another look? I've consolidated the menus to one file. I haven't fixed the issue where the full menu appears for runs, but I think I can tackle that when I move on to updating the nav state after login.

**Pull request merged**